src/lexer.rs

```diff
@@ -123,10 +118,6 @@ impl Lexer {
 
     /// char returns the current character
     fn char(&mut self) -> Option<char> {
-        self.line
-            .clone()
-            .unwrap_or_default()
-            .chars()
-            .nth(self.pos.try_into().unwrap_or_default())
+        self.line.as_ref().and_then(|l| l.chars().nth(self.pos))
```

Comment:
Remove `.clone` by creating a reference to the underlying option value and operating on that resulting in a temporary borrow instead of copying